### ISSUE TITLE
[PATCH] twisted.names.hosts not working correct with getHostByName

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7231]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| titty reported|
|-|-|
|Trac ID|trac#510|
|Type|defect|
|Created|2004-02-17 17:51:57Z|

<details><summary>Searchable metadata</summary>

```
trac-id__510 510
type__defect defect
reporter__titty titty
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1077040317000000 1077040317000000
changetime__1471666635132298 1471666635132298
version__ 
owner__ 
cc__radix cc__exarkun cc__spiv cc__titty
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|titty commented|
|-|-|

```
#!html
<pre>
Even with my prior patch I'm not able to use twisted.names.client.getHostByName.
Seems like twisted.names.hosts.Resolver calls itself with bad parameters
and even doesn't use the ip address it extracted from /etc/hosts. Also the return 
value could not be handled by the code in twisted.names.common.extractRecord.
The following patch seems to solve those problems.




===================================================================
RCS file: /cvs/Twisted/twisted/names/hosts.py,v
retrieving revision 1.14
diff -u -r1.14 hosts.py
--- twisted/names/hosts.py      19 Jul 2003 10:13:50 -0000      1.14
+++ twisted/names/hosts.py      17 Feb 2004 01:45:44 -0000
@@ -49,11 +49,11 @@
 
 
     def query(self, query, timeout = None):
-        return self._lookup(str(query.name), query.type, query.type, timeout)
+        return self._lookup(str(query.name), query.cls, query.type, timeout)
 
 
     def _lookup(self, name, cls, type, timeout):
-        if cls != dns.IN or type != dns.A:
+        if cls != dns.IN or (type != dns.A and type != dns.ALL_RECORDS): #
getHostByName passes type==dns.ALL_RECORDS. what else do we need here?
             return defer.fail(failure.Failure(NotImplementedError(type, cls)))
         return self.lookupAddress(
             name, timeout
@@ -64,7 +64,7 @@
         res = searchFileFor(self.file, name)
         if res:
             return defer.succeed([
-                [dns.RRHeader(name, dns.A, dns.IN, self.ttl)], (), ()
+                [dns.RRHeader(name, dns.A, dns.IN, self.ttl, dns.Record_A(res,
self.ttl))], [], []
             ])
         return defer.fail(failure.Failure(dns.DomainError(name)))
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
(By "prior patch", titty seems to mean #509)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
I skimmed the patch, but I don't think I'm familiar enough with DNS to tell
whether it's correct or not. Sorry :-(

exarkun, it's a pretty small patch. Can you take a look at it if you get some time?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Fixed in CVS, thanks titty. I didn't use your patch; hosts (and other minor
parts of t.names) were badly factored, and I fixed them up to make query and
_lookup unnecessary.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Fixed in CVS, thanks titty. I didn't use your patch; hosts (and other minor
parts of t.names) were badly factored, and I fixed them up to make query and
_lookup unnecessary.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Argh, had some trouble with the tracker. re-resolving.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc commented|
|-|-|

In changeset 6022f65903eda2f8df74b2d284b8ef359df24904
```
#!CommitTicketReference repository="" revision="6022f65903eda2f8df74b2d284b8ef359df24904"
Merge pull request #510 from twisted/8771-ssh-userauth-keyboard-interactive

Author: einarfd
Reviewer: rodrigc
Fixes: #8771

Fix interactive ssh prompt on Python 3
```
